main.js

```diff
@@ -33,25 +43,18 @@ app.disableHardwareAcceleration();
 // The PSP will have a single instance. If an attempt to start a second instance is made,
 // the second one will be closed and the callback provided to `app.makeSingleInstance`
 // in the first instance will be triggered enabling it to show the PSP `BrowserWindow`.
-var appIsRunning = app.makeSingleInstance(function (commandLine) {
+app.on("second-instance", function (event, commandLine) {
     var qssWrapper = fluid.queryIoCSelector(fluid.rootComponent, "gpii.app.qssWrapper")[0];
     qssWrapper.qss.show();
-
     if (commandLine.indexOf("--reset") > -1) {
-        process.nextTick(function () {
+        setTimeout(function () {
             // GPII-3455: Call this in the next tick, to allow electron to free some things.
```

Comment:
Please adjust this comment to reflect the fact that we seem to know that, for some reason, process.nextTick doesn't work any more (got even any vague guesses about why it doesn't?)

Reply:
Perhaps it has something to do with the way this callback is invoked. I looked at the electron code, and there's nothing obvious - the new process just sends a window message to the existing one, which raises the event. I've checked it's the same thread. Perhaps something is/isn't initialised above this invocation.

As to why it no longer works since Electron 3, perhaps the implementation of nextTick has changed slightly. Or maybe it's related to a change in edge.js. I just don't know.

I don't believe it's a time/delay issue - otherwise a zero delay wouldn't always work. But, if that turns out to be the case then we'll look into another way of implementing this rather than continue to chase a magic timing sweet spot.